doc/query.md

```diff
@@ -0,0 +1,545 @@
+    "pet": { "species": "cat" }
+}
+```
+
+To project the `name` and `age` fields of the above document:
+
+```java
+List<String> fields = new ArrayList<String>();
+fields.add("name");
+fields.add("age");
+QueryResult queryResult = im.find(query, 0, 0, fields, null);
+```
+
+Pass `null` as the `fields` argument to disable projection.
+
+#### Skip and limit
+
+Skip and limit allow retrieving subsets of the results. Amongst other things, this is useful in pagination.
+
+* `skip` skips over a number of results from the result set.
+* `limit` defines the maximum number of results to return for the query.
+
+To display the twenty-first to thirtieth results:
+
+```java
+QueryResult result = im.find(query, 20, 10, fields, null);
+```
+
+To disable:
+
+- `skip`, pass `0` as the `skip` argument.
+- `limit`, pass `0` as the `limit` argument.
+
+### Array fields
+
+Indexing and querying over array fields is supported in Cloudant Query Android, with some caveats.
+
+Take this document as an example:
+
+```
+{
+  _id: mike32
+  pet: [ cat, dog, parrot ],
+  name: mike,
+  age: 32
+}
+```
+
+You can create an index over the `pet` field:
+
+```java
+String name = im.ensureIndexed(Arrays.<Object>asList("name", "age", "pet"), 
+                               "basic");
+```
+
+Each value of the array is treated as a separate entry in the index. This means that a query such as:
+
+```
+{ pet: { $eq: cat } }
+```
+
+Will return the document `mike32`. Negation may be slightly confusing:
+
+```
+{ pet: { $not: { $eq: cat } } }
+```
+
+Will also return `mike32` because there are values in the array that are not `cat`. That is,
+this operator is matching for "any values that do not equal 'cat'".
+
+#### Restrictions
+
+Only one field in a given index may be an array. This is because each entry in each array
+requires an entry in the index, causing a Cartesian explosion in index size. Taking the
+above example, this document wouldn't be indexed because the `name` and `pet` fields are
+both indexed in a single index:
+
+
+```
+{
+  _id: mike32
+  pet: [ cat, dog, parrot ],
+  name: [ mike, rhodes ],
+  age: 32
+}
+```
+
+If this happens, an error will be emitted into the log but the indexing process will be
+successful.
+
+However, if there was one index with `pet` in and another with `name` in, like this:
+
+```java
+String indexOne = im.ensureIndexed(Arrays.<Object>asList("name", "age"), 
+                                   "index_one");
+String indexTwo = im.ensureIndexed(Arrays.<Object>asList("age", "pet"),
+                                   "index_two");
+```
+
+The document _would_ be indexed in both of these indexes: each index only contains one of
+the array fields.
+
+Also see "Unsupported features", below.
+
+
+### Errors
+
+Error reporting is terrible right now. The only indication something went wrong is a `null` return value from the `find` methods or `ensureIndexed(List<Object> fieldNames, String indexName)`. We're working on
+adding logging.
```

Comment:
Updated in https://github.com/cloudant/sync-android/commit/d5b4f7825035187c3ef55b5962f38b66806f68d3